modules/programs/firefox.nix

```
@@ -21,6 +21,11 @@ let
   profilesPath =
     if isDarwin then "${firefoxConfigPath}/Profiles" else firefoxConfigPath;
 
+  nativeMessagingHostsJoined = pkgs.symlinkJoin {
+    name = "home_ff_nmhs";
+    paths = cfg.package.nativeMessagingHosts or cfg.nativeMessagingHosts;
+  };
```

To support both, this could potentially include symlinks to `"${cfg.finalPackage}/lib/mozilla/native-messaging-host"` (though beware of bloat if you just add it to `paths`). Something like...

```suggestion
  nativeMessagingHostsJoined = pkgs.symlinkJoin {
    name = "home_ff_nmhs";
    paths = cfg.nativeMessagingHosts;
    inherit (cfg) finalPackage;
    postBuild = ''
      install -d $hostsPath
      for nativeHost in $finalPackage/$hostsPath/*; do
        ln -s $nativeHost $out/$hostsPath/
      fi
    '';
  };
```

Also the nixpkgs firefox wrapper does not appear to set `nativeMessagingHosts` in passthru or anything, so the `or` seems useless here?
Also should be using `cfg.finalPackage` if anything.